src/mscorlib/src/System/Threading/Mutex.cs

```diff
@@ -2,218 +2,123 @@
-                this.inCriticalRegion = inCriticalRegion;
+                throw new ArgumentException(SR.Format(SR.Argument_WaitHandleNameTooLong, Path.MaxPath), nameof(name));
             }
         }
 
-        public Mutex(bool initiallyOwned, String name) : this(initiallyOwned, name, out dummyBool)
-        {
-        }
-
-        public Mutex(bool initiallyOwned) : this(initiallyOwned, null, out dummyBool)
+        private void CreateMutexCore(bool initiallyOwned, string name, out bool createdNew)
         {
-        }
-
-        public Mutex() : this(false, null, out dummyBool)
-        {
-        }
-
-        private Mutex(SafeWaitHandle handle)
-        {
-            SetHandleInternal(handle);
-            hasThreadAffinity = true;
-        }
+            Debug.Assert(name == null || name.Length <= Path.MaxPath);
 
-        public static Mutex OpenExisting(string name)
-        {
-            return OpenExisting(name, (MutexRights)0);
-        }
+            uint mutexFlags = initiallyOwned ? Win32Native.CREATE_MUTEX_INITIAL_OWNER : 0;
 
-        internal enum MutexRights
-        {
-        }
+            SafeWaitHandle mutexHandle = Win32Native.CreateMutexEx(null, name, mutexFlags, AccessRights);
+            int errorCode = Marshal.GetLastWin32Error();
 
-        internal static Mutex OpenExisting(string name, MutexRights rights)
-        {
-            Mutex result;
-            switch (OpenExistingWorker(name, rights, out result))
+            if (mutexHandle.IsInvalid)
             {
-                case OpenExistingResult.NameNotFound:
-                    throw new WaitHandleCannotBeOpenedException();
-
-                case OpenExistingResult.NameInvalid:
+                mutexHandle.SetHandleAsInvalid();
+#if PLATFORM_UNIX
+                if (errorCode == Win32Native.ERROR_FILENAME_EXCED_RANGE)
+                    // On Unix, length validation is done by CoreCLR's PAL after converting to utf-8
+                    throw new ArgumentException(SR.Format(SR.Argument_WaitHandleNameTooLong, Interop.Sys.MaxName), nameof(name));
+#endif
+                if (errorCode == Interop.Errors.ERROR_INVALID_HANDLE)
                     throw new WaitHandleCannotBeOpenedException(SR.Format(SR.Threading_WaitHandleCannotBeOpenedException_InvalidHandle, name));
-
-                case OpenExistingResult.PathNotFound:
-                    throw Win32Marshal.GetExceptionForWin32Error(Win32Native.ERROR_PATH_NOT_FOUND, name);
-
-                default:
-                    return result;
+                throw Win32Marshal.GetExceptionForWin32Error(errorCode, name);
             }
-        }
 
-        public static bool TryOpenExisting(string name, out Mutex result)
-        {
-            return OpenExistingWorker(name, (MutexRights)0, out result) == OpenExistingResult.Success;
+            createdNew = errorCode != Interop.Errors.ERROR_ALREADY_EXISTS;
+            SafeWaitHandle = mutexHandle;
+            hasThreadAffinity = true;
```

Comment:
It looks like you have deleted this in other case, but not this one.

Reply:
Do you think it's safe to remove them everywhere now that I'm no longer using the `RuntimeHelpers.CleanupCode` method?

Reply:
It think so. It is also in WaitHandle and a few FCalls - it can be removed there as well.